src/data/polynomial/ring_division.lean

```diff
@@ -665,6 +665,32 @@ lemma root_set_finite (p : T[X])
   (S : Type*) [comm_ring S] [is_domain S] [algebra T S] : (p.root_set S).finite :=
 set.to_finite _
 
+/-- The set of roots of all polynomials of bounded degree and having coefficients in a finite set
+is finite. -/
+lemma bUnion_roots_finite {R S : Type*} [semiring R] [comm_ring S] [is_domain S]
+  (m : R →+* S) (d : ℕ) {U : set R} (h : U.finite) :
+  (⋃ (f : R[X]) (hf : f.nat_degree ≤ d ∧ ∀ i, (f.coeff i) ∈ U),
+    ((f.map m).roots.to_finset : set S)).finite :=
+begin
+  refine set.finite.bUnion _ _,
+  { -- We prove that the set of polynomials under consideration is finite because its
+    -- image by the injective map `π` is finite
+    let π : R[X] → finset.range (d+1) → R := λ f : R[X], λ i : finset.range (d+1), f.coeff i,
+    have h_inj : set.inj_on π  {f : R[X] | f.nat_degree ≤ d ∧ ∀ (i : ℕ), f.coeff i ∈ U},
```

Comment:
```suggestion
    let π : R[X] → finset.range (d+1) → R := λ f i, f.coeff i,
    have h_inj : set.inj_on π {f : R[X] | f.nat_degree ≤ d ∧ ∀ (i : ℕ), f.coeff i ∈ U},
```

Reply:
Actually since you didn't generalize as I suggested [here](https://github.com/leanprover-community/mathlib/pull/16670#discussion_r982647282) and stick to ℕ, using `fin (d+1)` probably makes the proof shorter. `finset.range` is just a way to write the set `{n | n < d+1}` such that Lean can infer it's finite; `fin (d+1)` is not a subset/subtype of ℕ (a recent change) but mathlib also knows it's finite and it's as easy to use as a subset/subtype.

Reply:
Yes, but I got intro (small) trouble using `fin (d+1)` since I could only cast `a : fin(d+1)` into a `nat` via `(a % (d+1) : nat)` whereas `finset.range` seems more flexible.

Reply:
I think `a % (d+1)` is for casting from nat to fin? Since we have [fin.fin_to_nat](https://leanprover-community.github.io/mathlib_docs/data/fin/basic.html#fin.fin_to_nat) you should be able just use a term of type `fin (d+1)` like a term of type `nat`.

Reply:
Oh, somehow, I missed this one.